opensrp-tasking/src/main/java/org/smartregister/tasking/util/GeoJsonUtils.java

```diff
@@ -0,0 +1,190 @@
+            mdaStatusMap.put(NOT_ELIGIBLE, 0);
+            mdaStatusMap.put(MDA_DISPENSE_TASK_COUNT, 0);
+            StateWrapper state = new StateWrapper();
+            if (taskSet == null)
+                continue;
+            for (Task task : taskSet) {
+                calculateState(task, state, mdaStatusMap);
+
+                taskProperties = new HashMap<>();
+                taskProperties.put(TASK_IDENTIFIER, task.getIdentifier());
+
+//                if (BuildConfig.BUILD_COUNTRY == Country.ZAMBIA && PARTIALLY_SPRAYED.equals(task.getBusinessStatus())) { // Set here for non residential structures
+//                    taskProperties.put(TASK_BUSINESS_STATUS, SPRAYED);
+//                } else {
+//                    taskProperties.put(TASK_BUSINESS_STATUS, task.getBusinessStatus());
+//                }
+                taskProperties.put(TASK_BUSINESS_STATUS, task.getBusinessStatus());
+
+                taskProperties.put(TaskingConstants.Properties.FEATURE_SELECT_TASK_BUSINESS_STATUS, task.getBusinessStatus()); // used to determine action to take when a feature is selected
+                taskProperties.put(TASK_STATUS, task.getStatus().name());
+                taskProperties.put(TaskingConstants.Properties.TASK_CODE, task.getCode());
+
+                if (indexCase != null && structure.getId().equals(indexCase)) {
+                    taskProperties.put(TaskingConstants.GeoJSON.IS_INDEX_CASE, Boolean.TRUE.toString());
+                } else {
+                    taskProperties.put(TaskingConstants.GeoJSON.IS_INDEX_CASE, Boolean.FALSE.toString());
+                }
+
+                taskProperties.put(TaskingConstants.Properties.LOCATION_UUID, structure.getProperties().getUid());
+                taskProperties.put(TaskingConstants.Properties.LOCATION_VERSION, structure.getProperties().getVersion() + "");
+                taskProperties.put(TaskingConstants.Properties.LOCATION_TYPE, structure.getProperties().getType());
+                interventionList.append(task.getCode());
+                interventionList.append("~");
+
+            }
+
+            populateBusinessStatus(taskProperties, mdaStatusMap, state);
+
+            taskProperties.put(TaskingConstants.Properties.TASK_CODE_LIST, interventionList.toString());
+            if (structureNames.get(structure.getId()) != null) {
+                taskProperties.put(Constants.DatabaseKeys.STRUCTURE_NAME, structureNames.get(structure.getId()).getStructureName());
+                taskProperties.put(TaskingConstants.Properties.FAMILY_MEMBER_NAMES, structureNames.get(structure.getId()).getFamilyMembersNames());
+            }
+            structure.getProperties().setCustomProperties(taskProperties);
+
+        }
+        return BaseInteractor.gson.toJson(structures);
+    }
+
+    public void calculateState(Task task, StateWrapper state, @NonNull Map<String, Integer> mdaStatusMap) {
+//        if (Utils.isResidentialStructure(task.getCode())) {
```

Comment:
I am not sure of the intention when commenting these out. Could you do something about it? Create an issue, todo, remove, fix